### PR TITLE
docs(env-variables): update docs to use T3 Env #1384

### LINKED ISSUE #1384: docs: rewrite env-variables docs page to use/reference new package
cba rewriting this now. creating an issue for it

_Originally posted by @juliusmarminge in https://github.com/t3-oss/create-t3-app/pull/1377#discussion_r1177128855_
            
Ref: https://env.t3.gg

### PR DESCRIPTION
**Pull request by esau-morais**
Closes #1384 

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)

---

## Changelog

_Rewrite docs for Environment Variables using T3 Env_ in English and Portuguese (Brazil)

---

💯

### CODE REVIEW

**Comment by esau-morais** [comment]
@juliusmarminge, do you mind checking if the "Validation Logic" section is necessary in the CT3 docs? Maybe we should focus everything related to T3 Env on its own docs.

Also, it's been a while since I've written posts/documentation, so I'd be really grateful to have some critical feedback about my writing

File: www/src/pages/pt/usage/env-variables.mdx [hunk 1]
```diff
@@ -1,130 +1,57 @@
...
+description: Iniciando com Create T3 App 
 layout: ../../../layouts/docs.astro
 lang: pt
 isMdx: true
 ---
 
 import Callout from "../../../components/docs/callout.tsx";
 
-O Create T3 App usa [Zod](https://github.com/colinhacks/zod) para validar suas variáveis de ambiente em runtime _e_ buildtime fornecendo alguns arquivos adicionais no diretório `env`:
+O Create T3 App usa sua mais nova biblioteca [@t3-oss/env-nextjs](https://env.t3.gg) e [zod](https://zod.dev) para validar variáveis de ambiente durante o tempo de execução (runtime) _e_ buildtime fornecendo uma lógica simples no arquivo `src/env.ts`:
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:11]:
```suggestion
O Create T3 App usa sua mais nova biblioteca [@t3-oss/env-nextjs](https://env.t3.gg) e [zod](https://zod.dev) para validar variáveis de ambiente durante o tempo de execução (runtime) _e_ buildtime fornecendo uma lógica simples no arquivo `src/env.mjs`:
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 2]
```diff
@@ -1,130 +1,57 @@
...
 
-## env.mjs
+## env.ts
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:13]:
```suggestion
## env.mjs
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 3]
```diff
@@ -1,130 +1,57 @@
...
 
-_Muito longo, não li; Se você quiser adicionar uma nova variável de ambiente, você deve adicioná-la tanto ao seu `.env` quanto definir o validador em `src/env.mjs`._
+_Muito longo, não li; Se você quiser adicionar uma nova variável de ambiente, você deve adicioná-la ao seu `.env` ao definir no validador em `src/env.ts`._
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:15]:
```suggestion
_Muito longo, não li; Se você quiser adicionar uma nova variável de ambiente, você deve adicioná-la ao seu `.env` ao definir no validador em `src/env.mjs`._
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 4]
```diff
@@ -1,130 +1,57 @@
...
 Este arquivo é dividido em duas partes - o esquema e a desestruturação do objeto, bem como a lógica de validação. A lógica de validação não deve ser tocada.
 
-```ts:env.mjs
-const server = z.object({
-  NODE_ENV: z.enum(["development", "test", "production"]),
-});
-
-const client = z.object({
-  // NEXT_PUBLIC_CLIENTVAR: z.string(),
+```ts:env.ts
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:19]:
```suggestion
```ts:env.mjs
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 5]
```diff
@@ -1,130 +1,57 @@
...
-  },
-});
-```
-
-</details>
-
-## Usando Variáveis Ambiente
-
-Quando você quiser usar suas variáveis de ambiente, você pode importá-las de `env.mjs` e usá-las como faria normalmente. Se você importar isso no cliente e tentar acessar uma variável de ambiente do lado do servidor, receberá um erro em tempo de execução.
+Quando você quiser usar suas variáveis de ambiente, você pode importá-las de `env.ts` (ou `env.mjs`) e usá-las como faria normalmente. Se você importar isso no cliente e tentar acessar uma variável de ambiente do lado do servidor, receberá um erro em tempo de execução (runtime).
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:44]:
```suggestion
Quando você quiser usar suas variáveis de ambiente, você pode importá-las de `env.mjs` e usá-las como faria normalmente. Se você importar isso no cliente e tentar acessar uma variável de ambiente do lado do servidor, receberá um erro em tempo de execução (runtime).
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 6]
```diff
@@ -1,130 +1,57 @@
...
 
 ```ts:pages/api/hello.ts
-import { env } from "../../env.mjs";
+import { env } from "../../env.ts";
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:47]:
```suggestion
import { env } from "../../env.mjs";
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 7]
```diff
@@ -1,130 +1,57 @@
...
 
-// `env` é totalmente tipado e fornece autocomplete
+// `env` é totalmente tipado e fornece preenchimento automático 
 const dbUrl = env.DATABASE_URL;
 ```
 
 ```ts:pages/index.tsx
-import { env } from "../env.mjs";
+import { env } from "../env.ts";
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:54]:
```suggestion
import { env } from "../env.mjs";
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 8]
```diff
@@ -145,34 +72,34 @@ Para garantir que sua compilação nunca seja concluída sem as variáveis de am
 
 📄 `.env`: Insira sua variável de ambiente como faria normalmente em um arquivo `.env`, ou seja, `CHAVE=VALOR`
 
-📄 `schema.mjs`: Adicione a lógica de validação apropriada para a variável de ambiente definindo um esquema Zod, por exemplo `TECLA: z.string()`
+📄 `env.ts`: Adicione a lógica de validação apropriada para as variáveis de ambiente definindo um esquema Zod, por exemplo `TECLA: z.string()`
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:75]:
```suggestion
📄 `env.mjs`: Adicione a lógica de validação apropriada para as variáveis de ambiente definindo um esquema Zod, por exemplo `NOME_DA_VARIAVEL: z.string()`
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 9]
```diff
@@ -145,34 +72,34 @@ Para garantir que sua compilação nunca seja concluída sem as variáveis de am
...
+_Quero adicionar meu token da API do Twitter como uma variável de ambiente do lado do servidor_
 
 1. Adicione a variável de ambiente a `.env`:
 
 ```
 TWITTER_API_TOKEN=1234567890
 ```
 
-2. Adicione a variável de ambiente a `schema.mjs`:
+2. Adicione a variável de ambiente a `env.ts`:
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:91]:
```suggestion
2. Adicione a variável de ambiente a `env.mjs`:
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 10]
```diff
@@ -145,34 +72,34 @@ Para garantir que sua compilação nunca seja concluída sem as variáveis de am
...
 
 ```ts
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:93]:
```suggestion
```ts:env.mjs
```

File: www/src/pages/pt/usage/env-variables.mdx [hunk 11]
```diff
@@ -145,34 +72,34 @@ Para garantir que sua compilação nunca seja concluída sem as variáveis de am
...
-export const serverSchema = z.object({
+import { createEnv } from "@t3-oss/env-nextjs";
+import { z } from "zod";
+ 
+export const env = createEnv({
+  server: {
+    TWITTER_API_TOKEN: z.enum(["development", "test", "production"]),
```
Comment (Sn0wye) [inline www/src/pages/pt/usage/env-variables.mdx:99]:
```suggestion
    TWITTER_API_TOKEN: z.string(),
    NODE_ENV: z.enum(["development", "test", "production"])
```

**Review by Sn0wye — CHANGES_REQUESTED** [review]
Nice writing throughout the whole file, but I'd suggest to maintain the `env.ts` file as `env.mjs`, since it runs before the compilation time, it cannot be Typescript.

**Review by Sn0wye — APPROVED** [review]
LGTM 🎉

File: www/src/pages/en/usage/env-variables.mdx [hunk 12]
```diff
@@ -8,119 +8,38 @@ isMdx: true
...
-  },
-});
-```
-
-</details>
-
 ## Using Environment Variables
 
-When you want to use your environment variables, you can import them from `env.mjs` and use them as you would normally do. If you import this on the client and try accessing a server-side environment variable, you will get a runtime error.
+When you want to use your environment variables, you can import them from the created `env.mjs` (or `env.mjs`) and use them as you would normally do. If you import this on the client and try accessing a server-side environment variable, you will get a runtime error.
```
Comment (juliusmarminge) [inline www/src/pages/en/usage/env-variables.mdx:42]: Double env.mjs

File: www/src/pages/en/usage/env-variables.mdx [hunk 13]
```diff
@@ -151,7 +70,7 @@ To ensure your build never completes without the environment variables the proje
 
 📄 `.env`: Enter your environment variable like you would normally do in a `.env` file, i.e. `KEY=VALUE`
 
-📄 `env.mjs`: Add the appropriate validation logic for the environment variable by defining a Zod schema, e.g. `KEY: z.string()`, and destruct the environment variable from `process.env` in the `processEnv` object, e.g. `KEY: process.env.KEY`.
+📄 `env.mjs`: Add the appropriate validation logic for the environment variables by defining a Zod schema inside `createEnv` for each one, e.g. `KEY: z.string()`.
```
Comment (juliusmarminge) [inline www/src/pages/en/usage/env-variables.mdx:73]: You still need to do destruction in the `runtimeEnv` option

File: www/src/pages/en/usage/env-variables.mdx [hunk 14]
```diff
@@ -170,15 +89,15 @@ TWITTER_API_TOKEN=1234567890
...
+  server: {
+    TWITTER_API_TOKEN: z.string(),
+  },
   // ...
-  TWITTER_API_TOKEN: z.string(),
 });
-
-export const processEnv = {
-  // ...
-  TWITTER_API_TOKEN: process.env.TWITTER_API_TOKEN,
```
Comment (juliusmarminge) [inline www/src/pages/en/usage/env-variables.mdx:180]: Put this in runtimeEnv

**Review by juliusmarminge — CHANGES_REQUESTED** [review]
Looks good - small nits

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**